### DISCUSSION TITLE
Install nlmix2r on Intel Mac

### DISCUSSION THREAD

**Discussion by BatheFu** [nlmixr2/nlmixr2#258] [Q&A]
I tried to install nlmix2r and rxode2 on Intel Macbook Pro 2019 model (Mac OS Sonoma 14.6), R version 4.2.3.

What I have done:
1. Install amendments of gfortran and OpenMP for Mac followed by https://mac.r-project.org/tools/.
2. Add gfortran directory to the path.
3. Type in codes.
`install.packages("nlmixr2",dependencies = TRUE)`

It failed but rxode2 (version 2.1.1 works with single thread)

Then I tried to install rxode2 from source
`install.packages("rxode2", type="source")`

`There are binary versions available but the source versions are later:
             binary source needs_compilation
rxode2et     2.0.11 2.0.13              TRUE
rxode2parse  2.0.17 2.0.19              TRUE
rxode2random 2.0.12  2.1.1              TRUE
qs           0.25.7 0.26.3              TRUE
rxode2        2.1.1  2.1.3              TRUE

...

* installing *source* package ‘qs’ ...
** package ‘qs’ successfully unpacked and MD5 sums checked
** using staged installation
checking for pkg-config... no
checking for clang++... clang++
checking whether the C++ compiler works... yes
checking for C++ compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C++ compiler... yes
checking whether clang++ accepts -g... yes
C++ compiler: clang++ -mmacosx-version-min=10.13 -std=gnu++14
...

ld: warning: -single_module is obsolete
ld: warning: -multiply_defined is obsolete
ld: library 'atomic' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [qs.so] Error 1
ERROR: compilation failed for package ‘qs’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/qs’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/qs’
Warning in install.packages :
  installation of package ‘qs’ had non-zero exit status

...

ld: warning: -single_module is obsolete
ld: warning: -multiply_defined is obsolete
ld: warning: ignoring duplicate libraries: '-lgfortran', '-lm', '-lquadmath'
ld: warning: search path '/usr/local/gfortran/lib/gcc/x86_64-apple-darwin18/8.2.0' not found
ld: warning: search path '/usr/local/gfortran/lib/gcc/x86_64-apple-darwin18/8.2.0' not found
ld: library 'gfortran' not found
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [rxode2.so] Error 1
ERROR: compilation failed for package ‘rxode2’
* removing ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/rxode2’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.2/Resources/library/rxode2’
Warning in install.packages :
  installation of package ‘rxode2’ had non-zero exit status
`
I emitted some rows with ...

I was confused with the last lines, because I installed gfortran darwin20, but it warned me with darwin18.

I also runned the OpenMP testing code mentioned [here](https://github.com/nlmixr2/nlmixr2/discussions/26)

And the result is,
omp_test()
OpenMP not supported

What should I do next?
Thanks.

**Comment by BatheFu** [comment]
I retrieved the version of nlmixr2 to 2.0.9 and solved it.

**Reply by mattfidler** [reply]
I'm glad it was resolved!
